src/main/java/Ui.java

```diff
@@ -0,0 +1,147 @@
+                    msg += "     " + count + "." + t.toString() + "\n";
+                }
+                count++;
+            }
+            System.out.println(this.format(msg));
+            return tasks;
+        } else if (parser.getTaskType().equals("done")) {
+            int count = 1;
+            ArrayList<Task> taskList = tasks.getList();
+            String msg = "     Nice! I've marked this task as done:\n";
+            int taskDone = parser.getTaskIdx();
+            for (int i = 0; i < taskList.size(); ++i) {
+                if (i == taskDone - 1) {
+                    taskList.set(i, taskList.get(i).markAsDone());
+                    msg += "     " + taskList.get(i);
+                }
+            }
+            System.out.println(this.format(msg));
+            return new TaskList(taskList);
+        } else if (parser.getTaskType().equals("delete")) {
+            int taskToDelete = parser.getTaskIdx();
+            String msg = "     Noted. I've removed this task: \n";
+            msg += "     " + tasks.get(taskToDelete - 1) + "\n";
+            tasks.remove(taskToDelete - 1);
+            msg += "     Now you have " + tasks.size() + " tasks in the list.";
+            System.out.println(this.format(msg));
+            return tasks;
+        } else if (parser.getTaskType().equals("find")) {
+            String toFind = parser.getDescription();
+            ArrayList<Task> taskList = tasks.getList();
+            String msg = "     Here are the matching tasks in your list:\n";
+            boolean first = true;
+            for (int i = 0; i < taskList.size(); ++i) {
+                if (taskList.get(i).getDescription().contains(toFind)) {
+                    if (first) {
+                        msg += "     " + (i + 1) + "." + taskList.get(i).toString();
+                        first = false;
+                    } else {
+                        msg += "\n     " + (i + 1) + "." + taskList.get(i).toString();
+                    }
+                }
+            }
+            if (first) {
+                msg = "     There are no matching tasks";
+            }
+            System.out.println(this.format(msg));
+            return tasks;
+        } else if (parser.getTaskType().equals("todo")) {
+            String msg = "     Got it. I've added this task:\n";
+            String taskDescription = parser.getDescription();
+            Todo todo = new Todo(taskDescription);
+            tasks = tasks.add(todo);
+            msg += "     " + todo + "\n";
+            msg += "     Now you have " + tasks.size() + " tasks in the list.";
+            System.out.println(this.format(msg));
+        } else if (parser.getTaskType().equals("deadline")) {
+            String msg = "     Got it. I've added this task:\n";
+            LocalDate dueDate = parser.getDueDate();
+            String taskDescription = parser.getDescription();
+            Deadline deadline = new Deadline(taskDescription, dueDate);
```

Comment:
Since you do not plan to use dueDate or taskDescription, could you directly place parser.getDueDate() and parser.getDescription() in the constructor instead?